### PR TITLE
tgswitch: use buildGoModule

### PR DESCRIPTION
**Pull request by aaronjheng**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/cluster/tgswitch/default.nix [hunk 1]
```diff
@@ -10,7 +10,11 @@ buildGoPackage rec {
     sha256 = "sha256-6hErfI7LEJFgOoJR8IF9jTSBwqbQYeGiwdeJShqxVQ0=";
   };
 
-  goPackagePath = "github.com/warrensbox/tgswitch";
+  vendorSha256 = null;
+
+  ldflags = [ "-s" "-w" ];
+
+  doCheck = false;
```
Comment (psibi) [inline pkgs/applications/networking/cluster/tgswitch/default.nix:17]: Is there any specific on why the tests are disable ? Can you add a comment for the same ?

File: pkgs/applications/networking/cluster/tgswitch/default.nix [hunk 2]
```diff
@@ -10,7 +10,11 @@ buildGoPackage rec {
     sha256 = "sha256-6hErfI7LEJFgOoJR8IF9jTSBwqbQYeGiwdeJShqxVQ0=";
   };
 
-  goPackagePath = "github.com/warrensbox/tgswitch";
+  vendorSha256 = null;
+
+  ldflags = [ "-s" "-w" ];
```
Comment (psibi) [inline pkgs/applications/networking/cluster/tgswitch/default.nix:15]: Can you add comments about these ldflags ? I guess `-s` is for stripping the symbols. not sure about `-w`.

File: pkgs/applications/networking/cluster/tgswitch/default.nix [hunk 3]
```diff
@@ -1,5 +1,5 @@
-{ buildGoPackage, lib, fetchFromGitHub }:
-buildGoPackage rec {
+{ buildGoModule, lib, fetchFromGitHub }:
+buildGoModule rec {
```
Comment (psibi) [inline pkgs/applications/networking/cluster/tgswitch/default.nix:2]: Haven't been following the buildGoModule vs buildGoPackage conversation. Is the recommended practice now to use buildGoModule ?

**Review by psibi — CHANGES_REQUESTED** [review]
LGTM, some minor comments. Thanks!

File: pkgs/applications/networking/cluster/tgswitch/default.nix (hunk 2 above)
Comment (aaronjheng) [reply pkgs/applications/networking/cluster/tgswitch/default.nix:15]: These are common flags used in nixpkgs for smaller binary size. See https://pkg.go.dev/cmd/link.

File: pkgs/applications/networking/cluster/tgswitch/default.nix (hunk 3 above)
Comment (aaronjheng) [reply pkgs/applications/networking/cluster/tgswitch/default.nix:2]: It's marked as legacy now.

Per nixpkgs manual:
> The function buildGoPackage builds legacy Go programs, not supporting Go modules.

File: pkgs/applications/networking/cluster/tgswitch/default.nix (hunk 1 above)
Comment (aaronjheng) [reply pkgs/applications/networking/cluster/tgswitch/default.nix:17]: `buildGoPackage` disable checks by default.
```
doCheck = args.doCheck or false;
```

with `doCheck` enabled, there are some failed tests. So I disable checks for now.

Comment (aaronjheng) [reply pkgs/applications/networking/cluster/tgswitch/default.nix:17]: Maybe I can try to fix it.

File: pkgs/applications/networking/cluster/tgswitch/default.nix (hunk 2 above)
Comment (psibi) [reply pkgs/applications/networking/cluster/tgswitch/default.nix:15]: Ah, got it. Thanks!

File: pkgs/applications/networking/cluster/tgswitch/default.nix (hunk 3 above)
Comment (psibi) [reply pkgs/applications/networking/cluster/tgswitch/default.nix:2]: Thanks!

File: pkgs/applications/networking/cluster/tgswitch/default.nix (hunk 1 above)
Comment (psibi) [reply pkgs/applications/networking/cluster/tgswitch/default.nix]: If you are not able to fix it, I guess you can just a comment explaining why the tests are failing... LGTM apart from this single comment. Thanks!

**Review by juboba — APPROVED** [review]

**Review by psibi — APPROVED** [review]
Thank you!

**Comment by aaronjheng** [comment]
I have no idea why there is a `/var/empty` directory on macOS.

**Comment by aaronjheng** [comment]
Looks like these tests use `user.HomeDir` other than `os.UserHomeDir`.

**Comment by aaronjheng** [comment]
I do not think it's not worth to enable tests for now.

**Comment by psibi** [comment]
@aaronjheng Yeah, agree.  Probably just disable the tests and add a comment. :-)

**Review by psibi — APPROVED** [review]

**Pull request merged**